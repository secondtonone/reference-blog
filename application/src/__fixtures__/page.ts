import {
  PageViewModel
} from '~/view-model';
import fixtureArticle from '~/__fixtures__/article-content';

const page: PageViewModel = {
  id: 1,
  previewImage: '/previewImage.jpg',
  title: 'title',
  author: {
    id: 111,
    name: 'Olga Testova',
    alias: 'olga-testova',
    email: 'example@example.com',
    position: 'Super seo specialist',
    company: 'Yandex',
    bio:
      'Olga Testova is a Super seo specialist at <a href="http://yandex.ru">Yandex</a>, where she develops SEO and content strategies for ecommerce and technology clients. Her career dream is to make SEO more accessible for marketers of all experience levels. She believes wholeheartedly that better search results are better for everyone.',
    photo: 'https://storage.googleapis.com/semblog-rc/testing/user/olga.jpg',
    socials: {
      webSite: 'https://example.com',
      twitter: 'https://twitter.com/hello',
      facebook: '100009766004824',
      instagram: 'hello.hello',
      linkedin: 'chto-chto-942423146',
    },
    isPublic: true
  },
  publishedAt: new Date('2020-01-01').getTime(),
  category: {
    id: 1,
    name: 'name',
    slug: 'slug',
    lang: 'lang',
    children: null
  },
  timeToRead: 1,
  url: 'url',
  template: 'article-base',
  socialImage: 'socialImage',
  preview: 'preview',
  extraImage: 'extra.png',
  altPreviewImage: 'Preview Image',
  altSocialImage: 'Social Image',
  hidePreviewImage: 0,
  editedAt: new Date('2020-02-02').getTime(),
  metaTitle: 'metaTitle',
  metaDescription: 'metaDescription',
  schemaOrg: 'schemaOrg',
  related: [5916, 5915, 5914, 5913],
  translated: null,
  critical: {
    'article-content': {
      id: 1,
      content: fixtureArticle,
      team: 'team',
      galaxy: 'galaxy'
    }
  },
  alternates: [],
  status: 'published',
  tableOfContent: true
};

export default page;
