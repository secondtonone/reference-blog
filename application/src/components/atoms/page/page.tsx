import styled from 'styled-components';

const Page = (props) => (
  <PageStyled data-test="page" {...props} />
);

const PageStyled = styled.div<{ thin?: boolean }>`
  position: relative;
  font-family: 'Inter', Ubuntu, Helvetica, Arial, sans-serif;
  width: 100%;
  height: 100%;
  min-height: ${({ thin }) => (!thin ? '100vh' : '100%')};
  display: flex;
  flex-direction: column;
  will-change: background-color, color;
  min-width: 0;
`;

export default Page;
