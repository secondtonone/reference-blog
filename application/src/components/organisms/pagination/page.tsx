import { IButtonProps } from '@semcore/button';
import ButtonLink from '~/components/atoms/button-link';
import PaginationLink from './pagination-link';

interface Props extends IButtonProps {
  selected?: boolean,
  number: number,
  onPageClick?: (page: number) => void
  dataTest?: string
}

const Page: React.FC<Props> = ({
  number,
  selected = false,
  onPageClick,
  'data-test': dataTest,
  children,
}) => (
  <PaginationLink page={number}>
    {(url) => (
      <ButtonLink
        data-test={dataTest || 'pagination-page'}
        href={url}
        use="secondary"
        selected={selected}
        className="pagination-page"
        style={{ minWidth: 42 }}
        onClick={(e) => {
          if (onPageClick instanceof Function) {
            onPageClick(number);
          }
          e.preventDefault();
        }}
      >
        {children || number}
      </ButtonLink>
    )}
  </PaginationLink>
);

export default Page;
