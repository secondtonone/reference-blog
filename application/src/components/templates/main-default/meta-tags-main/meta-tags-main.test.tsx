import renderer from 'react-test-renderer';
import page from '~/__fixtures__/page';
import MetaTagsMain from '.';

const Component: React.FC = ({ children }) => <>{children}</>;

jest.mock('next/head', () => ({
  __esModule: true,
  default: Component
}));

const alternates = [{ language: 'es', url: 'blog/perspectiva-digital-juegos-olimpicos-2020/', title: 'La perspectiva digital de los Juegos Ol\u00EDmpicos de 2020: cifras y hechos clave' }, { language: 'it', url: 'blog/olimpiadi-2020-prospettiva-digitale/', title: 'La prospettiva digitale delle Olimpiadi 2020: statistiche e fatti chiave' }, { language: 'pt', url: 'blog/perspectiva-digital-olimpiadas-2020/', title: 'Perspectiva Digital das Olimp\u00EDadas 2020: principais estat\u00EDsticas e fatos' }, { language: 'en', url: 'blog/olympics-2020-digital-perspective/', title: 'The Olympics 2020 Digital Perspective: Key Stats and Facts' }].map(({ language, url }) => [language, url]);

describe('meta tags main renders correctly', () => {
  it('with preview top article', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={1}
        previewCard={page}
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with two preview big articles', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={1}
        bigCards={[page, { ...page, previewImage: '/previewImage1.jpg', }]}
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 1 with invalid en domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={1}
        alternates={alternates}
        // @ts-ignore
        canonicalSubdomain="en"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 1 with default domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={1}
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 2 and en domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={2}
        canonicalSubdomain="www"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 3 and es domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={3}
        canonicalSubdomain="es"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 4 and de domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={4}
        canonicalSubdomain="de"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 5 and fr domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={5}
        canonicalSubdomain="fr"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 6 and it domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={6}
        canonicalSubdomain="it"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('by page 7Ñ‡ and pt domain', () => {
    const tree = renderer.create(
      <MetaTagsMain
        currentPage={7}
        canonicalSubdomain="pt"
        alternates={alternates}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
