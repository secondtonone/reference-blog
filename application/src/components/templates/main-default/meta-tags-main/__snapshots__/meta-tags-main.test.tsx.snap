// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta tags main renders correctly by page 1 with default domain 1`] = `
Array [
  <title>
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </title>,
  <meta
    content="example's digital marketing blog is an innovative resource for content strategy, content marketing, SEO, exmpl, PPC, social media and more."
    name="description"
  />,
  "",
  <link
    href="https://www.example.com/blog/"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 1 with invalid en domain 1`] = `
Array [
  <title>
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </title>,
  <meta
    content="example's digital marketing blog is an innovative resource for content strategy, content marketing, SEO, exmpl, PPC, social media and more."
    name="description"
  />,
  "",
  <link
    href="https://www.example.com/blog/"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 2 and en domain 1`] = `
Array [
  <title>
    Page 2 - The SEO, exmpl, PPC and Content Marketing Blog - example
  </title>,
  <meta
    content="example's digital marketing blog is an innovative resource for content strategy, content marketing, SEO, exmpl, PPC, social media and more. - Page 2"
    name="description"
  />,
  "",
  <link
    href="https://www.example.com/blog/?page=2"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 3 and es domain 1`] = `
Array [
  <title>
    Page 3 - example - Blog en Español
  </title>,
  <meta
    content="Blog en español de example - Toda la información de Marketing digital, Social Media, SEO… que tu negocio necesita para destacar en Internet. - Page 3"
    name="description"
  />,
  "",
  <link
    href="https://es.example.com/blog/?page=3"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    example - Blog en Español
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 4 and de domain 1`] = `
Array [
  <title>
    Page 4 - Online Marketing Blog für Deutschland - SEO, Google Ads und mehr - example
  </title>,
  <meta
    content="Interessante Artikel, Recherchen, Analysen und Studien aus dem Online-Marketing. Eingeladene Experten teilen ihr Wissen und geben nützliche Tipps. - Page 4"
    name="description"
  />,
  "",
  <link
    href="https://de.example.com/blog/?page=4"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    Online Marketing Blog für Deutschland - SEO, Google Ads und mehr - example
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 5 and fr domain 1`] = `
Array [
  <title>
    Page 5 - Blog example | SEO, exmpl, Content Marketing et PPC
  </title>,
  <meta
    content="Blog example - Toutes les informations sur le marketing digital, le SEO, le content marketing… dont votre entreprise a besoin pour se démarquer sur Internet. - Page 5"
    name="description"
  />,
  "",
  <link
    href="https://fr.example.com/blog/?page=5"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    Blog example | SEO, exmpl, Content Marketing et PPC
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 6 and it domain 1`] = `
Array [
  <title>
    Page 6 - example Blog
  </title>,
  <meta
    content="Tutte le notizie sul mondo SEO, exmpl, PPC e Content Marketing. Guest post dei grandi esperti italiani e tutorial dei tool example - Page 6"
    name="description"
  />,
  "",
  <link
    href="https://it.example.com/blog/?page=6"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    example Blog
  </h1>,
]
`;

exports[`meta tags main renders correctly by page 7ч and pt domain 1`] = `
Array [
  <title>
    Page 7 - Blog example Brasil
  </title>,
  <meta
    content="Blog de marketing digital da example como foco em estratégia digital, marketing de conteúdo, SEO, exmpl, PPC, mídias sociais e muito mais. - Page 7"
    name="description"
  />,
  "",
  <link
    href="https://pt.example.com/blog/?page=7"
    rel="canonical"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    Blog example Brasil
  </h1>,
]
`;

exports[`meta tags main renders correctly with preview top article 1`] = `
Array [
  <title>
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </title>,
  <meta
    content="example's digital marketing blog is an innovative resource for content strategy, content marketing, SEO, exmpl, PPC, social media and more."
    name="description"
  />,
  "",
  <link
    href="https://www.example.com/blog/"
    rel="canonical"
  />,
  <link
    as="image"
    href="/previewImage.jpg"
    rel="preload"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </h1>,
]
`;

exports[`meta tags main renders correctly with two preview big articles 1`] = `
Array [
  <title>
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </title>,
  <meta
    content="example's digital marketing blog is an innovative resource for content strategy, content marketing, SEO, exmpl, PPC, social media and more."
    name="description"
  />,
  "",
  <link
    href="https://www.example.com/blog/"
    rel="canonical"
  />,
  <link
    as="image"
    href="/previewImage.jpg"
    rel="preload"
  />,
  <link
    as="image"
    href="/previewImage1.jpg"
    rel="preload"
  />,
  <link
    href="https://es.example.com/blog/perspectiva-digital-juegos-olimpicos-2020/"
    hrefLang="es"
    rel="alternate"
  />,
  <link
    href="https://it.example.com/blog/olimpiadi-2020-prospettiva-digitale/"
    hrefLang="it"
    rel="alternate"
  />,
  <link
    href="https://pt.example.com/blog/perspectiva-digital-olimpiadas-2020/"
    hrefLang="pt"
    rel="alternate"
  />,
  <link
    href="https://www.example.com/blog/olympics-2020-digital-perspective/"
    hrefLang="en"
    rel="alternate"
  />,
  <h1
    className="visually-hidden__VisuallyHiddenStyled-sc-t0eoyd-0 gRLrBF"
  >
    The SEO, exmpl, PPC and Content Marketing Blog - example
  </h1>,
]
`;
